backend/sirius-web-diagrams-layout/src/main/java/org/eclipse/sirius/web/diagrams/layout/LayoutService.java

```diff
@@ -82,7 +85,47 @@ public LayoutService(ELKDiagramConverter elkDiagramConverter, IncrementalLayoutD
     @Override
     public Diagram layout(IEditingContext editingContext, Diagram diagram) {
         ELKConvertedDiagram convertedDiagram = this.elkDiagramConverter.convert(diagram);
+        return layoutInternal(editingContext, diagram, convertedDiagram);
+    }
+
+    @Override
+    public Node layoutNode(IEditingContext editingContext, Diagram diagram, Node node) {
+        ELKConvertedDiagram convertedDiagram = this.elkDiagramConverter.convert(diagram, node);
+        Diagram layoutedDiagram = layoutInternal(editingContext, diagram, convertedDiagram);
+
+        return layoutedDiagram.getNodes()
+                .stream()
+                .filter(layoutedDiagramNode -> layoutedDiagramNode.getId().equals(node.getId()))
+                .findFirst().orElse(null);
+    }
 
+    @Override
+    public Diagram incrementalLayout(IEditingContext editingContext, Diagram newDiagram, Optional<IDiagramEvent> optionalDiagramElementEvent) {
```

Comment:
Please don't move code around like that without a good reason, it makes it more difficult to understand what you have changed.